flake.nix

```diff
@@ -0,0 +1,51 @@
+{
+  description = "Simple, modern libpng alternative";
+
+  inputs = {
+    utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:NixOS/nixpkgs";
+    libspng-src = {
+      url = "github:randy408/libspng";
+      flake = false;
+    };
+  };
+
+  outputs = { self, nixpkgs, utils, libspng-src }:
+    utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages."${system}";
+      in
+      rec {
+        # `nix build`
+        packages.libspng = pkgs.stdenv.mkDerivation {
+          pname = "libspng";
+          name = "libspng";
```

Comment:
`pname` and `name` should not be used at the same time.
Either just:
 - `name`
 - or `pname` and `version`
 
 pname and version should always be preferred instead of `name`.